Helper/PaymentMethods.php

```diff
@@ -133,41 +138,51 @@ public function getPaymentMethods($quoteId, $country = null)
     {
         // get quote from quoteId
         $quote = $this->quoteRepository->getActive($quoteId);
-        $store = $quote->getStore();
 
-        $paymentMethods = $this->fetchAlternativeMethods($store, $country);
+        // If quote cannot be found early return the empty paymentMethods array
+        if (empty($quote)) {
+            return [];
+        }
+
+        $this->setQuote($quote);
+
+        $paymentMethods = $this->fetchAlternativeMethods($country);
         return $paymentMethods;
     }
 
     /**
-     * @param $store
      * @param $country
      * @return array
      */
-    protected function fetchAlternativeMethods($store, $country)
+    protected function fetchAlternativeMethods($country)
     {
-        $merchantAccount = $this->adyenHelper->getAdyenAbstractConfigData('merchant_account');
+        $quote = $this->getQuote();
```

Comment:
why not doing a $this->quote instead ?

Reply:
The getQuote() getter was already in the code and used several times. Instead of replacing it with the $this->quote I decided to keep using it and add the setter as well.